snapcraft/parts/lifecycle.py

```diff
@@ -449,7 +448,7 @@ def _run_in_provider(
             capture_logs_from_instance(instance)
         except subprocess.CalledProcessError as err:
             capture_logs_from_instance(instance)
-            raise ProviderError(
+            raise errors.SnapcraftError(
```

Comment:
Fix incorrectly used exception, `ProviderError` is the exception used by craft-providers to report errors during instance creation and operation.